operator/pkg/apis/validation/validation.go

```diff
@@ -0,0 +1,390 @@
+		validateGateways(spec.Components.EgressGateways, "egress")
+	}
+
+	return warnings, nil
+}
+
+// checkServicePorts validates Service ports. Specifically, this currently
+// asserts that all ports will bind to a port number greater than 1024 when not
+// running as root.
+func checkServicePorts(values *apis.Values, spec apis.IstioOperatorSpec) (Warnings, util.Errors) {
+	var errs util.Errors
+	if spec.Components != nil {
+		if !values.GetGateways().GetIstioIngressgateway().GetRunAsRoot().GetValue() {
+			errs = util.AppendErrs(errs, validateGateways(spec.Components.IngressGateways, "istio-ingressgateway"))
+		}
+		if !values.GetGateways().GetIstioEgressgateway().GetRunAsRoot().GetValue() {
+			errs = util.AppendErrs(errs, validateGateways(spec.Components.EgressGateways, "istio-egressgateway"))
+		}
+	}
+	for _, raw := range values.GetGateways().GetIstioIngressgateway().GetIngressPorts() {
+		p := raw.AsMap()
+		var tp int
+		if p["targetPort"] != nil {
+			t, ok := p["targetPort"].(float64)
+			if !ok {
+				continue
+			}
+			tp = int(t)
+		}
+
+		rport, ok := p["port"].(float64)
+		if !ok {
+			continue
+		}
+		portnum := int(rport)
+		if tp == 0 && portnum > 1024 {
+			// Target port defaults to port. If its >1024, it is safe.
+			continue
+		}
+		if tp < 1024 {
+			// nolint: lll
+			errs = util.AppendErr(errs, fmt.Errorf("port %v is invalid: targetPort is set to %v, which requires root. Set targetPort to be greater than 1024 or configure values.gateways.istio-ingressgateway.runAsRoot=true", portnum, tp))
+		}
+	}
+	return nil, errs
+}
+
+func validateGateways(gws []apis.GatewayComponentSpec, name string) util.Errors {
+	// nolint: lll
+	format := "port %v/%v in gateway %v invalid: targetPort is set to %d, which requires root. Set targetPort to be greater than 1024 or configure values.gateways.%s.runAsRoot=true"
+	var errs util.Errors
+	for _, gw := range gws {
+		if gw.Kubernetes == nil || gw.Kubernetes.Service == nil {
+			continue
+		}
+		for _, p := range gw.Kubernetes.Service.Ports {
+			tp := 0
+			if p.TargetPort.Type == intstr.String {
+				// Do not validate named ports
+				continue
+			}
+			if p.TargetPort.Type == intstr.Int {
+				tp = int(p.TargetPort.IntVal)
+			}
+			if tp == 0 && p.Port > 1024 {
+				// Target port defaults to port. If its >1024, it is safe.
+				continue
+			}
+			if tp < 1024 {
+				errs = util.AppendErr(errs, fmt.Errorf(format, p.Name, p.Port, gw.Name, tp, name))
+			}
+		}
+	}
+	return errs
+}
+
+func validateHub(hub string) error {
+	if hub == "" {
+		return nil
+	}
+	// Takes a <hub>/image, we have <hub>, so pass an image we would use
+	_, err := name.NewRepository(hub + "/pilot")
+	if err != nil {
+		return fmt.Errorf("bad hub: %v", err)
+	}
+	return nil
+}
+
+func validateTag(tag any) error {
+	if tag == nil {
+		return nil
+	}
+	// Takes a <hub>/<image>:<tag>, we just have <tag>
+	_, err := name.NewTag(fmt.Sprintf("istio/pilot:%v", tag))
```

Comment:
Nice - I am happy to use `"github.com/google/go-containerregistry/pkg/name"` here for the validation opinions.